zkvm/src/vm.rs

```diff
@@ -306,6 +306,22 @@ where
         Ok(())
     }
 
+    fn mul(&mut self) -> Result<(), VMError> {
+        let expr2 = self.pop_item()?.to_expression()?;
+        let expr1 = self.pop_item()?.to_expression()?;
+        let expr3 = expr1.multiply(expr2, self.delegate.cs());
+        self.push_item(expr3);
+        Ok(())
+    }
+
+    fn eq(&mut self) -> Result<(), VMError> {
+        let expr2 = self.pop_item()?.to_expression()?;
+        let expr1 = self.pop_item()?.to_expression()?;
+        let expr3 = expr1 - expr2;
```

Comment:
See `Constraint::Eq` type. We don't need to compute difference between `Expression`s. This will happen at r1cs level within the implementation of `Constraint` (inside `verify` instruction).